lib/rubocop/cop/rails/active_support_on_load.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Use Active Support lazy load hooks to patch Rails core classes, so they are not forcibly loaded early.
+      #
+      # @example
+      #
+      #   # bad
+      #   ActiveRecord::Base.include(MyClass)
+      #
+      #   # good
+      #   ActiveSupport.on_load(:active_record) { include MyClass }
+      class ActiveSupportOnLoad < Base
+        extend AutoCorrector
+
+        AUTOCORRECTABLE_CLASSES = {
```

Comment:
I think it will be a name that matches the L37 local variable better:

```suggestion
        HOOK_NAMES = {
```